src/hooks/core/push-notification/usePushNotificationHook.ts

```diff
@@ -0,0 +1,218 @@
+          index: screenData ? 2 : 1,
+          routes
+        });
+      }
+    }, 500);
+  };
+
+  const __handleNotification = async (notification) => {
+    if (notification.data.type === 'feed' || notification.data.type === 'reaction') {
+      navigation.navigate('PostDetailPage', {
+        feedId: notification.data.feed_id,
+        is_from_pn: true
+      });
+    }
+    if (notification.data.type === 'follow_user') {
+      navigation.navigate('OtherProfile', {
+        data: {
+          user_id: notification.data.user_id,
+          other_id: notification.data.user_id_follower,
+          username: notification.data.username_follower
+        }
+      });
+    }
+    if (notification.data.type === 'message.new') {
+      if (notification?.data?.receiver_id === signedProfileId) {
+        const channel = new ChannelList({
+          id: notification?.data?.channel_id,
+          channelType: 'PM'
+        });
+        setIsLoadingFetchingChannelDetail(true);
+        try {
+          await fetchChannelDetail(channel);
+          const selectedChannel = await ChannelList.getSchemaById(
+            localDb,
+            notification?.data?.channel_id
+          );
+          setSelectedChannel(selectedChannel);
+          helperNavigationResetWithData({
+            screen: 'SignedChatScreen'
+          });
+        } catch (e) {
+          console.log('error', e);
+        } finally {
+          setIsLoadingFetchingChannelDetail(false);
+        }
+      }
+    }
+  };
+
+  React.useEffect(() => {
+    __createChannel();
+    __updateProfileAtomId();
+
+    messaging().setBackgroundMessageHandler(async (remoteMessage) => {
+      __handlePushNotif(remoteMessage);
+    });
+
+    const unsubscribe = messaging().onMessage((remoteMessage) => {
+      // eslint-disable-next-line no-unused-expressions
+      __handlePushNotif(remoteMessage);
+    });
+    return () => {
+      unsubscribe();
+    };
+  }, []);
+
+  React.useEffect(() => {
+    PushNotification.configure({
+      // (required) Called when a remote is received or opened, or local notification is opened
+      onNotification(notification) {
+        __handleNotification(notification);
+        notification.finish(PushNotificationIOS.FetchResult.NoData);
+      },
+      // (optional) Called when the user fails to register for remote notifications.
+      // Typically occurs when APNS is having issues, or the device is a simulator. (iOS)
+      onRegistrationError(err) {
+        console.error(err.message, err);
+      },
+      // IOS ONLY (optional): default: all - Permissions to register.
+      permissions: {
+        alert: true,
+        badge: true,
+        sound: true
+      },
+      // Should the initial notification be popped automatically
+      // default: true
+      popInitialNotification: true,
+      requestPermissions: false
+    });
+
+    return () => {
+      PushNotification.unregister();
+    };
+  }, [navigation, signedProfileId, localDb]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the second useEffect hook, you are passing `navigation`, `signedProfileId`, and `localDb` as dependencies but they are not being used in the effect. This could lead to unnecessary re-renders. Consider removing them from the dependency array.

```diff
-  }, [navigation, signedProfileId, localDb]);
+  }, []);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->